src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect } from 'react';
+import { DataSourceSelector } from './datasource_selector';
+import { DataSourceType } from '../datasource_services';
+import { DataSourceGroup, DataSourceOption, DataSourceSelectableProps } from './types';
+import { ISourceDataSet } from '../datasource/types';
+import { IndexPattern } from '../../index_patterns';
+
+// Mapping between datasource type and its display name.
+const DATASOURCE_TYPE_DISPLAY_NAME_MAP = {
```

Comment:
Instead of hardcoding the mapping like this, cant we simply have the name as a part of the datasource registration api?

Reply:
I actually thought about it -> yes but at the same time it may worth discussing about what are the information we should add to datasource which may contain data more related to backend. in this case here, it's actually what the type name should be for datasource vs what should be the type name we display in datasource selector. When initializes datasource we may still call a s3glue as type name somewhere, but in selector it should be called as Amazon s3 let's just say. So it would have this additional level of mapping for now until we have a more clear path for definition of all these name, types, metadata. Then it should be more easily to remove this and modify datasource

Reply:
True, but we can incrementally expand that. Id start with:

```ts
// datasource type
{
  name: string,
  label: string,
}
```

e.g.
```ts
registerDataSourceType({
  name: 'index_patterns',
  label: i18n.translate('data.datasource.type.indexPattern', { 
    defaultMessage: 'Index Patterns' 
  }),
}, dataSourceClass);
```

Reply:
Additionally this can also be added to the datasource class if necessary

Reply:
Will address after this PR merge, Tracking [issue](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5211).